woocommerce/woocommerce-android
# Issue/5870 save inbox notes

⚠️ Please do not merge until this FluxC [PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2335) is merged.

Closes: #5870


### Description
In this PR `InboxViewmodel` fetches notes from API and observes for DB changes to refresh the notes list. 

### Testing instructions
Open Inbox section from More Menu and check that the list is loaded and refreshed if needed. 

### Images/gif
As you can see from the recording, sometimes there is a flicker when displaying the Inbox notes list. I've tried to fix this in several ways but haven't found a solution to it. Basically, there are several emissions of the same list items when entering the screen. Not sure why this happens as the `observeInboxNotes` method from WCInboxStore in FLuxC applies `.distinctUntilChanged()` to the Flow chain, so events that are equal should not be emitted.

**EDIT**: The flickering issue was solved after fixing several issues on this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2335) which these changes rely on. 

https://user-images.githubusercontent.com/2663464/161005748-9925e43b-8e85-4128-b25b-85f98781aa8a.mp4

Will keep looking into this in a different issue as I don't want to block this task more time due to this minor issue.
